modules/kubernetes/Makefile

```diff
@@ -0,0 +1,231 @@
+ervice on cluster
+LUSTER_NAMESPACE ?= 
+CLUSTER_DOMAIN ?= mertslounge.ca
+CLUSTER_BASTION ?= bastion.$(CLUSTER_NAMESPACE).$(CLUSTER_DOMAIN)
+export CLUSTER_NAMESPACE
+  
+KUBERNETES_NAMESPACE ?= sagan
+KUBERNETES_CONTEXT ?= aws_kubernetes
+KUBERNETES_NAME ?= $(subst -docker,,$(shell basename "`pwd`"))
+KUBERNETES_RESOURCE_PATH ?= ./kubernetes
+export KUBERNETES_NAME
+
+# Kubectl specific settings
+KUBECTL ?= /opt/bin/kubectl
+KUBECTL_SCHEMA_CACHE_DIR ?= --schema-cache-dir=.kube/cache
+KUBECTL_SSH_CMD := ssh -A -o 'BatchMode=yes' -o 'LogLevel=error' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null'
+# Optionally define an SSH command to use for tunneling kubectl commands
+KUBECTL_SSH_TUNNEL ?= $(KUBECTL_SSH_USER)@$(CLUSTER_BASTION)
+
+# Serial number optionally used for versioning; select the last 5 characters of unix time (24 character limit to resource names in k8s)
+SERIAL ?= $(shell echo -n $$(date +%s) | tail -c 5)
+export SERIAL
+
+define kubectl_create
+	@echo -e "INFO: Creating $(KUBERNETES_NAME) $(1) on cluster $(call yellow,$(CLUSTER_NAMESPACE))..."
+	@envsubst < "$(KUBERNETES_RESOURCE_PATH)/$(KUBERNETES_NAME)-$(1).yml" | envsubst | $(KUBECTL_CMD) create $(KUBECTL_SCHEMA_CACHE_DIR) -f -
+endef
+
+define kubectl_delete
+	@echo -e "INFO: Deleteting $(KUBERNETES_NAME) $(1) on cluster $(call yellow,$(CLUSTER_NAMESPACE))..."
+	@envsubst < "$(KUBERNETES_RESOURCE_PATH)/$(KUBERNETES_NAME)-$(1).yml" | envsubst | $(KUBECTL_CMD) delete --ignore-not-found=true -f -
+endef
+
+ifeq ($(CIRCLECI),true)
+  KUBECTL_SSH_CMD += -i '$(HOME)/.ssh/id_circleci_github'
+  KUBECTL_SSH_USER ?= saganbot
+  CLUSTER_NAMESPACE ?= $(CIRCLE_BRANCH)
+endif
+
+ifeq ($(strip $(KUBECTL_SSH_USER)),)
+  # Guess their github username
+	KUBECTL_SSH_USER = $(shell ssh git@github.com 2>&1 | grep 'successfully authenticated' | cut -d' ' -f2 | cut -d'!' -f1)
+endif
+
+# Add the SSH endpoint to the command; everything after the host is expected to be a remote command
+KUBECTL_SSH_CMD += $(KUBECTL_SSH_TUNNEL)
+
+KUBECTL_CMD ?= $(KUBECTL_SSH_CMD) "$(KUBECTL)" --logtostderr=true --insecure-skip-tls-verify=true  
+
+# Set the namespace if one exists
+ifneq ($(KUBERNETES_NAMESPACE),)
+  KUBECTL_CMD += --namespace="$(KUBERNETES_NAMESPACE)" 
+endif
+
+# Set the URL to the kubernetes master, if one exists
+ifneq ($(KUBERNETES_URL),)
+  KUBECTL_CMD += --server="$(KUBERNETES_URL)" 
+endif
+
+# Set the user to authenticate against Kubernetes, if one exists
+ifneq ($(KUBERNETES_USER),)
+  KUBECTL_CMD += --username="$(KUBERNETES_USER)" 
+endif
+
+# Set the password to authenticate against Kubernetes, if one exists
+ifneq ($(KUBERNETES_PASS),)
+  KUBECTL_CMD += --password="$(KUBERNETES_PASS)" 
+endif
+
+DOCKERCFG_FILE ?= .dockercfg
+
+ifndef KUBERNETES_NAMESPACE
+  $(error KUBERNETES_NAMESPACE is undefined)
+endif
+
+export KUBERNETES_NAMESPACE
+
+#
+# Reference Docs:
+#  - https://cloud.google.com/container-engine/docs/kubectl/
+#
+## Display info about the kubernetes setup
```

Comment:
there are so many targets here, and we only use a fraction of them, what do you think of thinning this out or removing them from help to keep the list clearer?

99 times out of 100 we only need to use `deploy`

Alternatively we could separate the logic for deploying, targets like `deploy/delete` and the setup targets like `up/down`.  Keep the deploy stuff here because its tied to building, and either leave the up/down in the kube-harness or move it into cloud-platform where it is mostly likely to be used.